### PR TITLE
Added org.eclipse.elk.alg.libavoid

### PR DESCRIPTION
**Pull request by spoenemann**
This PR adds a new plug-in `org.eclipse.elk.alg.libavoid` that integrates the [libavoid](https://www.adaptagrams.org/documentation/libavoid.html) library with ELK. The plug-in is provided to the update site in a separate feature because libavoid is LGPL licensed, so not all users might want to include it.

The license check for this new dependency is approved:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=24155

The Java code is taken from https://github.com/TypeFox/elk-libavoid, which in turn has been copied from the [KIELER](https://rtsys.informatik.uni-kiel.de/confluence/display/KIELER/Home) project by the rtsys group at Kiel University.

The Maven build downloads three binaries from https://github.com/TypeFox/libavoid-server/releases/tag/v0.1.0 which contain the actual routing algorithm. This needs to be done for testing the algorithm in the development workspace as well. I'm not sure what would be the best solution for that... could it be added to the Oomph setup?

### CODE REVIEW

**Comment by spoenemann** [comment]
I created [IPLab request #4846](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4846). Please wait for approval before merging this PR.

**Comment by spoenemann** [comment]
Update: The IPLab request is approved. @soerendomroes do you have some time to look through this contribution?

**Review by soerendomroes — COMMENTED** [review]
Do you have example graphs to test this algorithm with?

File: plugins/org.eclipse.elk.alg.libavoid/src/org/eclipse/elk/alg/libavoid/LibavoidLayoutProvider.java
Comment (spoenemann) [reply plugins/org.eclipse.elk.alg.libavoid/src/org/eclipse/elk/alg/libavoid/LibavoidLayoutProvider.java:51]: I have not touched the actual implementation, it's still in the original state. I propose to address issues and todos of the implementation separately. This PR is just for bringing it into the ELK code base.

**Comment by spoenemann** [comment]
> Do you have example graphs to test this algorithm with?

```
algorithm: libavoid

node n1 {
	layout [
		position: 20, 20
		size: 30, 30
	]
	portConstraints: FIXED_POS
	port p1 {
		layout [
			position: 30, 10
			size: 5, 5
		]
		side: EAST
	}
	port p2 {
		layout [
			position: 30, 20
			size: 5, 5
		]
		side: EAST
	}
}

node n2 {
	layout [
		position: 80, 20
		size: 30, 30
	]
	portConstraints: FIXED_POS
	port p3 {
		layout [
			position: -5, 12
			size: 5, 5
		]
		side: WEST
	}
}

node n3 {
	layout [
		position: 150, 30
		size: 30, 30
	]
	portConstraints: FIXED_POS
	port p4 {
		layout [
			position: -5, 12
			size: 5, 5
		]
		side: WEST
	}
}

edge n1.p1 -> n2.p3
edge n1.p2 -> n3.p4
```

**Comment by soerendomroes** [comment]
Do you plan to do an Elk Blogpost about libavoid? It also be a good place to explain the different licenses.

**Review by soerendomroes — APPROVED** [review]

**Comment by spoenemann** [comment]
Yes, I'll add a post about this.

@soerendomroes what do you think regarding the dev setup?

> The Maven build downloads three binaries from https://github.com/TypeFox/libavoid-server/releases/tag/v0.1.0 which contain the actual routing algorithm. This needs to be done for testing the algorithm in the development workspace as well. I'm not sure what would be the best solution for that... could it be added to the Oomph setup?

**Comment by spoenemann** [comment]
Re: dev setup &ndash; the easiest might be adding a shell script that downloads the binaries.

**Pull request merged**
